OneSignal/OneSignal-iOS-SDK
# Add additional 6 privacy manifests to the 6 sub-targets

# Description
## One Line Summary
Add additional privacy manifests to User, Notifications, LiveActivities, Outcomes, OSCore, and Core by following [Apple documentation](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files?language=objc).

## Details
In https://github.com/OneSignal/OneSignal-iOS-SDK/pull/1334 we added Privacy Manifest to the 4 targets that clients would import. 

There was this comment in the PR:

> Currently, they are not added to the other targets such as Core, Outcomes, User, etc. as these targets are already accounted for by inclusion in the umbrella `OneSignalFramework` target. App developers do not import these smaller modules directly but only via `OneSignalFramework`.
>
> **We will await feedback if the above is true and these "sub-modules" don't need a privacy manifest.**

We haven't received concrete information on the last question, but since that PR, Apple has released a list of [SDKs that require a privacy manifest and signature](https://developer.apple.com/support/third-party-SDK-requirements/) and listed the following player model SDKs, leading me to believe they do detect every sub target?
- OneSignal
- OneSignalCore
- OneSignalExtension
- OneSignalOutcomes

### Motivation
Third-party privacy manifests required by Apple.

### Scope
Privacy Manifests

# Testing
## Unit testing
None

## Manual testing
Build and archive app, inspect the Privacy Report
**See the generated report for the example app** [OneSignalExample-PrivacyReport.pdf](https://github.com/OneSignal/OneSignal-iOS-SDK/files/15279229/OneSignalExample-PrivacyReport.pdf)

# Affected code checklist

# Checklist
## Overview

## Testing

## Final pass


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/OneSignal-iOS-SDK/1431)
